### PR TITLE
feat(specs): Large preimage proposals

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds specs to the monorepo for large preimage proposals (https://www.notion.so/oplabs/Keccak-Dispute-5e068b79375d454faedc6ed999925ee3?pvs=4)

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/90

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9035** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9035?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9039** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9039?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #9052** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9052?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9052?utm_source=stack-comment).

File: specs/fault-dispute-game.md [hunk 1]
```diff
@@ -255,12 +256,70 @@ The FDG provides the following interface to manage data loaded to the `PreimageO
...
+```
+
+Once the full preimage and all intermediate state commitments have been posted, the large preimage proposal enters a
+challenge period. During this time, a challenger can reconstruct the merkle tree that was progressively built on-chain
+locally by scanning the block bodies that contain the proposer's leaf preimages. If they detect that a commitment to
+the intermediate state of the hash function is incorrect at any step, they may perform a single-step dispute for the
+proposal in the `PreimageOracle`. This involves:
+
+1. Creating a merkle proof for the agreed upon prestate leaf (not necessary if the invalid leaf is the first one, the
+   setup state of the matrix is constant.) within the proposal's merkle root.
```
Comment (ajsutton) [inline specs/fault-dispute-game.md:313]:
```suggestion
   setup state of the matrix is constant) within the proposal's merkle root.
```

**Review by ajsutton — COMMENTED** [review]
Spec update looks good but there's a bunch of extra changes in the PR not related to the spec.  Ideally we'd avoid adding the additional space characters through unrelated parts of the spec too - not quite sure where they came from.

**Comment by clabby** [comment]
@ajsutton rebased, got rid of other files 👍 

The changes in spaces look to have come from a linter, but look correct. Seems to have added indents for text in a bullet point and trimmed leading whitespace on newlines.

**Review by ajsutton — APPROVED** [review]

**Pull request merged**